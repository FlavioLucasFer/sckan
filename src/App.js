import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import { 
  LOGIN_ROUTE, 
  COMPANY_ROUTE, 
  PROJECT_ROUTE, 
  SPRINT_ROUTE, 
  TASK_ROUTE, 
  USER_ROUTE,  
} from 'core/utils/routes';

import Login from 'features/login/Login';
import Company from 'features/company/Company';
import Project from 'features/project/Project';
import Sprint from 'features/sprint/Sprint';
import Task from 'features/task/Task';
import User from 'features/user/User';

function App() {
  return (
    <Router>
      <Route exact path={LOGIN_ROUTE} component={Login} />
      <Route path={COMPANY_ROUTE} component={Company} />
      <Route path={PROJECT_ROUTE} component={Project} />
      <Route path={SPRINT_ROUTE} component={Sprint} />
      <Route path={TASK_ROUTE} component={Task} />
      <Route path={USER_ROUTE} component={User} />
    </Router>
  );
}

export default App;
