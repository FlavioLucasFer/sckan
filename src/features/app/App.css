
#main {
	padding-top: 20px;
	overflow-y: auto !important;
}

#sidenav {
	padding: 0;
	height: calc(100vh - 50px) !important;
	overflow-y: auto;
}