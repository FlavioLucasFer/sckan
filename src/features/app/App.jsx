import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import {
	LOGIN_ROUTE,
	COMPANY_ROUTE,
	PROJECT_ROUTE,
	SPRINT_ROUTE,
	TASK_ROUTE,
	USER_ROUTE,
} from 'core/utils/routes';

import Header from 'features/app/components/Header';

import Login from 'features/login/Login';
import Company from 'features/company/Company';
import Project from 'features/project/Project';
import Sprint from 'features/sprint/Sprint';
import Task from 'features/task/Task';
import User from 'features/user/User';

function App() {
	return (
		<Router data-testid='router'>
			<Header data-testid='header' />

			<Route exact path={LOGIN_ROUTE} component={Login} data-testid='login-screen-route' />
			<Route path={COMPANY_ROUTE} component={Company} data-testid='companies-screen-route' />
			<Route path={PROJECT_ROUTE} component={Project} data-testid='projects-screen-route' />
			<Route path={SPRINT_ROUTE} component={Sprint} data-testid='sprints-screen-route' />
			<Route path={TASK_ROUTE} component={Task} data-testid='tasks-screen-route' />
			<Route path={USER_ROUTE} component={User} data-testid='users-screen-route' />
		</Router>
	);
}

export default App;
