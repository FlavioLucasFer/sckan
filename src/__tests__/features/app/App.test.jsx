import React from 'react';
import { shallow } from 'enzyme';

import App from 'features/app/App';

describe('Testing App screen component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<App />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});
});