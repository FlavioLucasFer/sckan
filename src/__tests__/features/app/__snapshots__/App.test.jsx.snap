// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App screen component should render correctly 1`] = `
"<BrowserRouter data-testid=\\"router\\">
  <Route path=\\"/login\\" component={[Function: Login]} data-testid=\\"login-screen-route\\" />
  <Header data-testid=\\"header\\" />
  <div className=\\"row\\" style={{...}}>
    <nav id=\\"sidenav\\" className=\\"col s0 m3 l2 white\\" style={{...}}>
      <SideNavBar />
    </nav>
    <main id=\\"main\\" className=\\"col s12 m9 l10\\">
      <div className=\\"container\\">
        <Route path=\\"/\\" component={[Function: App]}>
          <Route exact={true} path=\\"/\\" component={[Function: Home]} data-testid=\\"home-screen-route\\" />
          <Route path=\\"/companies\\" data-testid=\\"companies-screen-route\\">
            <Route exact={true} path=\\"/companies\\" component={[Function: Companies]} data-testid=\\"companies-screen-route\\" />
            <Route exact={true} path=\\"/companies/form\\" component={[Function: CompanyForm] { propTypes: { id: [Function: bound checkType] { isRequired: [Function: bound checkType] } } }} data-testid=\\"company-form-route\\" />
          </Route>
          <Route path=\\"/users\\" data-testid=\\"users-screen-route\\">
            <Route exact={true} path=\\"/users\\" component={[Function: Users]} data-testid=\\"users-screen-route\\" />
            <Route exact={true} path=\\"/users/form\\" component={[Function: UserForm] { propTypes: { id: [Function: bound checkType] { isRequired: [Function: bound checkType] } } }} data-testid=\\"users-screen-route\\" />
          </Route>
          <Route path=\\"/projects\\" data-testid=\\"projects-screen-route\\">
            <Route exact={true} path=\\"/projects\\" component={[Function: Projects]} data-testid=\\"projects-screen-route\\" />
            <Route exact={true} path=\\"/projects/form\\" component={[Function: ProjectForm] { propTypes: { id: [Function: bound checkType] { isRequired: [Function: bound checkType] } } }} data-testid=\\"project-form-screen-route\\" />
          </Route>
          <Route path=\\"/sprints\\" data-testid=\\"sprints-screen-route\\">
            <Route exact={true} path=\\"/sprints\\" component={[Function: Sprints]} data-testid=\\"sprints-screen-route\\" />
            <Route exact={true} path=\\"/sprints/form\\" component={[Function: SprintForm] { propTypes: { id: [Function: bound checkType] { isRequired: [Function: bound checkType] } } }} data-testid=\\"sprints-form-screen-route\\" />
          </Route>
          <Route path=\\"/tasks\\" data-testid=\\"tasks-screen-route\\">
            <Route exact={true} path=\\"/tasks\\" component={[Function: Task]} data-testid=\\"tasks-screen-route\\" />
          </Route>
        </Route>
      </div>
    </main>
  </div>
</BrowserRouter>"
`;
